### PR TITLE
Update UPAPI for Dedicated Gateway in Self Serve Portal

### PR DESCRIPTION
**Pull request by tarazou9**
# Description
This is for updating today's UPAPI with the new one for Dedicated Gateway in Self Serve Portal. 

# Changes
Replacing today's UPAPI calls with two API calls. 
- First API call is to get Offering Ids for each sku by service name and location. Here is the reference link: https://learn.microsoft.com/en-us/rest/api/marketplacecatalog/dataplane/skus/list?view=rest-marketplacecatalog-dataplane-2023-05-01-preview&tabs=HTTP
- Second API call is making to the new UPAPI to use Offering Id for each sku to retrieve prices. 

# How Has This Been Tested?
Test locally from end to end.

### CODE REVIEW

**Review by asierisayas — DISMISSED** [review]

File: src/Common/Constants.ts [hunk 1]
```diff
@@ -249,6 +249,7 @@ export class HttpHeaders {
   public static partitionKey: string = "x-ms-documentdb-partitionkey";
   public static migrateOfferToManualThroughput: string = "x-ms-cosmos-migrate-offer-to-manual-throughput";
   public static migrateOfferToAutopilot: string = "x-ms-cosmos-migrate-offer-to-autopilot";
+  public static xAPIKey: string = "X-API-Key";
```
Comment (jawelton74) [inline src/Common/Constants.ts:252]: should the value be lower case ?

File: src/ConfigContext.ts [hunk 2]
```diff
@@ -42,6 +42,9 @@ export interface ConfigContext {
   ARM_API_VERSION: string;
   GRAPH_ENDPOINT: string;
   GRAPH_API_VERSION: string;
+  CATALOG_ENDPOINT: string;
```
Comment (jawelton74) [inline src/ConfigContext.ts:45]: Please add a comment about what this endpoint is

File: src/Common/Constants.ts (hunk 1 above)
Comment (tarazou9) [reply src/Common/Constants.ts:252]: @jawelton74 Here is the API document link, it looks like it's using "X-API-Key" in the doc. https://learn.microsoft.com/en-us/rest/api/marketplacecatalog/dataplane/skus/list?view=rest-marketplacecatalog-dataplane-2023-05-01-preview&tabs=HTTP#request-headers

File: src/ConfigContext.ts (hunk 2 above)
Comment (tarazou9) [reply src/ConfigContext.ts:46]: @jawelton74 Comment added, thank you for pointing this out!

**Review by jawelton74 — APPROVED** [review]

**Review by asierisayas — APPROVED** [review]

**Pull request merged**